src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -601,6 +595,28 @@ private Element GenerateLicenseElement(string licenseInfo)
         return licenseElement;
     }
 
+    private SpdxEntities.Relationship GetSpdxRelationship(string sourceElement, string targetElement, SbomEntities.RelationshipType relationshipType)
+    {
+        var spdxRelationshipType = this.GetSPDXRelationshipType(relationshipType);
+
+        // Switch source and target IDs for these specific relationship types
+        if (relationshipType == SbomEntities.RelationshipType.PREREQUISITE_FOR ||
```

Comment:
It feels odd to do this for SPDX 3 but not SPDX 2. Did SPDX redefine the direction of the relationship between the two versions?

Reply:
In SPDX 2.2 there were some relationships that the provided inverses for as well. For example, DESCRIBES and DESCRIBED_BY were inverses of each other. 

In 3.0, these inverses are no longer present. So in the above example, only DESCRIBES is valid. Therefore if we get a DESCRIBED_BY relationship we must convert it to DESCRIBES by changing the directionality for it to be a valid SPDX 3.0 relationship type.

Reply:
Ah. Looking at `GetSPDXRelationshipType`, I now see where we map `PREREQUISITE_FOR`, `DESCRIBED_BY`, and `PATCH_FOR` to `HAS_PREREQUISITE`, `DESCRIBES`, and `PATCHED_BY`, respectively. I completely missed that subtlety until now. That feels like it deserves a comment--maybe something like this inside the switch block on line 631?

```
// These 3 relationships intentionally change the relationship direction to conform with the SPDX 3.0 spec
case SbomEntities.RelationshipType.PREREQUISITE_FOR: return RelationshipType.HAS_PREREQUISITE;
case SbomEntities.RelationshipType.DESCRIBED_BY: return RelationshipType.DESCRIBES;
case SbomEntities.RelationshipType.PATCH_FOR: return RelationshipType.PATCHED_BY;
```